pydantic/networks.py

```diff
@@ -86,136 +88,294 @@ def __hash__(self) -> int:
+            if annotated_type := schema['type'] not in ('url', 'multi-host-url'):
+                raise PydanticUserError(
+                    f"'{cls.__name__}' cannot annotate '{annotated_type}'.", code='invalid-annotated-type'
+                )
+            for constraint_key, constraint_value in cls._constraints.set_constraints.items():
+                schema[constraint_key] = constraint_value
+            return schema
 
-* TLD not required
-* Host required
-"""
-WebsocketUrl = Annotated[Url, UrlConstraints(max_length=2083, allowed_schemes=['ws', 'wss'])]
-"""A type that will accept any ws or wss URL.
 
-* TLD not required
-* Host required
-* Max length 2083
-"""
-FileUrl = Annotated[Url, UrlConstraints(allowed_schemes=['file'])]
-"""A type that will accept any file URL.
+class _BaseMultiHostUrl(MultiHostUrl):
+    _constraints: ClassVar[UrlConstraints] = UrlConstraints()
 
-* Host not required
-"""
-FtpUrl = Annotated[Url, UrlConstraints(allowed_schemes=['ftp'])]
-"""A type that will accept ftp URL.
+    @classmethod
+    def __get_pydantic_core_schema__(cls, source: type[Any], handler: GetCoreSchemaHandler) -> core_schema.CoreSchema:
+        if cls is source:
+            return core_schema.multi_host_url_schema(**cls._constraints.set_constraints)
+        else:
+            schema = handler(source)
+            # TODO: this logic is used in types.py as well in the _check_annotated_type function, should we move that to somewhere more central?
+            if annotated_type := schema['type'] not in ('url', 'multi-host-url'):
+                raise PydanticUserError(
+                    f"'{cls.__name__}' cannot annotate '{annotated_type}'.", code='invalid-annotated-type'
+                )
+            for constraint_key, constraint_value in cls._constraints.set_constraints.items():
+                schema[constraint_key] = constraint_value
+            return schema
+
+
+class AnyUrl(_BaseUrl):
+    """Base type for all URLs.
+
+    * Any scheme allowed
+    * Top-level domain (TLD) not required
+    * Host required
+
+    Assuming an input URL of `http://samuel:pass@example.com:8000/the/path/?query=here#fragment=is;this=bit`,
+    the types export the following properties:
+
+    - `scheme`: the URL scheme (`http`), always set.
+    - `host`: the URL host (`example.com`), always set.
+    - `username`: optional username if included (`samuel`).
+    - `password`: optional password if included (`pass`).
+    - `port`: optional port (`8000`).
+    - `path`: optional path (`/the/path/`).
+    - `query`: optional URL query (for example, `GET` arguments or "search string", such as `query=here`).
+    - `fragment`: optional fragment (`fragment=is;this=bit`).
+    """
 
-* TLD not required
-* Host required
-"""
-PostgresDsn = Annotated[
-    MultiHostUrl,
-    UrlConstraints(
+    _constraints = UrlConstraints(host_required=True)
```

Comment:
Probably this fixes https://github.com/pydantic/pydantic/issues/10684, maybe a test can confirm

Reply:
You're correct, I think the removal of that `file` test from the parametrized `AnyUrl` test shows this fix.